src/detector/implementations/picamera2_implementation.py

```diff
@@ -0,0 +1,181 @@
+from src.detector.api import \
+    GetCaptureDeviceResponse, \
+    GetCaptureImageRequest, \
+    GetCaptureImageResponse, \
+    GetCapturePropertiesResponse, \
+    SetCaptureDeviceRequest, \
+    SetCapturePropertiesRequest
+from src.common import \
+    EmptyResponse, \
+    ErrorResponse, \
+    get_kwarg, \
+    MCastResponse
+from src.common.structures.capture_status import CaptureStatus
+
+from src.detector.implementations import AbstractCameraInterface
+
+from picamera2 import Picamera2, Preview
+
+import base64
+import cv2
+import datetime
+import logging
+import time
+import numpy
+import os
+from typing import Any, Callable
+
+logger = logging.getLogger(__name__)
+
+class PiCamera(AbstractCameraInterface):
+
+    _capture: numpy.ndarray | None
+    _captured_image: numpy.ndarray | None
+
+    def __init__(self):
+        self._capture = None
+        self._captured_image = None
+        self._captured_timestamp_utc = datetime.datetime.min
+
+        self._capture_status = CaptureStatus
+        self._capture_status.status = CaptureStatus.Status.STOPPED
+
+        self._captured_timestamp_utc = datetime.datetime.min
+
+        self._capture_status = CaptureStatus
+        self._capture_status.status = CaptureStatus.Status.STOPPED
+
+        # TODO: DEBUGGING
+        self._capture_status.status = CaptureStatus.Status.RUNNING
```

Comment:
still needed?

Reply:
I am not sure, I copied it from the original code. I'm not entirely sure what its purpose is.

Reply:
Interestingly, it seems obsolete in the USB webcam implementation, but is still necessary in the pi camera one.

Reply:
What happens if you remove the line of code? Does starting the camera work as intended?

Reply:
The camera starts and an initial frame is shown but it does not update.

Reply:
and are you referring to the GUI?

Reply:
My apologies, I should be more clear. If I remove the line then the camera still starts and connects to the GUI just fine, but after the initial frame appears in the GUI the image is never updated. And yes, I am not changing my usage of the .start() command specific to the Picamera2 python library, nor the AbstractCameraInterface.start_capture() method.

Reply:
This is now fixed on my machine with my latest commit.